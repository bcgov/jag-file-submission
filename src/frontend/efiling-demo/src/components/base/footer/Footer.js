import React from "react";

export const LinkElement = (url, label) => (
  <li className="nav-item m-1">
    <a
      className="nav-link"
      href={url}
      target="_blank"
      rel="noopener noreferrer"
    >
      {label}
    </a>
  </li>
);

export const FooterToggler = () => (
  <button
    className="navbar-toggler"
    type="button"
    data-toggle="collapse"
    data-target="#footerBar"
  >
    <span className="navbar-toggler-icon" />
  </button>
);

export const Footer = () => (
  <footer className="footer">
    <nav className="navbar navbar-expand-sm navbar-dark justify-content-end">
      <FooterToggler />
      <div className="collapse navbar-collapse flex-grow-0" id="footerBar">
        <ul className="navbar-nav text-right">
          {LinkElement("https://www2.gov.bc.ca", "BC Government")}
          {LinkElement(
            "https://www2.gov.bc.ca/gov/content/home/disclaimer",
            "Disclaimer"
          )}
          {LinkElement(
            "https://www2.gov.bc.ca/gov/content/home/privacy",
            "Privacy"
          )}
          {LinkElement(
            "https://www2.gov.bc.ca/gov/content/home/accessibility",
            "Accessibility"
          )}
          {LinkElement(
            "https://www2.gov.bc.ca/gov/content/home/copyright",
            "Copyright"
          )}
        </ul>
      </div>
    </nav>
  </footer>
);
