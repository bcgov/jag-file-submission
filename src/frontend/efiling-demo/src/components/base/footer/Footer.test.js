import React from "react";
import testBasicSnapshot from "../../../TestHelper";

import { Footer, LinkElement } from "./Footer";

describe("Footer Component", () => {
  test("Matches the snapshot", () => {
    testBasicSnapshot(<Footer />);
  });

  test("Link element renders the link and matches the snapshot", () => {
    const linkElement = LinkElement("example.com", "Example");

    testBasicSnapshot(linkElement);
  });
});
