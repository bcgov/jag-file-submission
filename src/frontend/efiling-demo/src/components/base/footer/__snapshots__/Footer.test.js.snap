// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Link element renders the link and matches the snapshot 1`] = `
<li
  className="nav-item m-1"
>
  <a
    className="nav-link"
    href="example.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    Example
  </a>
</li>
`;

exports[`Footer Component Matches the snapshot 1`] = `
<footer
  className="footer"
>
  <nav
    className="navbar navbar-expand-sm navbar-dark justify-content-end"
  >
    <button
      className="navbar-toggler"
      data-target="#footerBar"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse flex-grow-0"
      id="footerBar"
    >
      <ul
        className="navbar-nav text-right"
      >
        <li
          className="nav-item m-1"
        >
          <a
            className="nav-link"
            href="https://www2.gov.bc.ca"
            rel="noopener noreferrer"
            target="_blank"
          >
            BC Government
          </a>
        </li>
        <li
          className="nav-item m-1"
        >
          <a
            className="nav-link"
            href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
            rel="noopener noreferrer"
            target="_blank"
          >
            Disclaimer
          </a>
        </li>
        <li
          className="nav-item m-1"
        >
          <a
            className="nav-link"
            href="https://www2.gov.bc.ca/gov/content/home/privacy"
            rel="noopener noreferrer"
            target="_blank"
          >
            Privacy
          </a>
        </li>
        <li
          className="nav-item m-1"
        >
          <a
            className="nav-link"
            href="https://www2.gov.bc.ca/gov/content/home/accessibility"
            rel="noopener noreferrer"
            target="_blank"
          >
            Accessibility
          </a>
        </li>
        <li
          className="nav-item m-1"
        >
          <a
            className="nav-link"
            href="https://www2.gov.bc.ca/gov/content/home/copyright"
            rel="noopener noreferrer"
            target="_blank"
          >
            Copyright
          </a>
        </li>
      </ul>
    </div>
  </nav>
</footer>
`;
